tests/admin_views/test_autocomplete_view.py

```diff
@@ -0,0 +1,173 @@
+        request = factory.get(self.url, {'field_identifier': 'admin_views.Answer.question', 'term': 'is'})
+        self.user.is_staff = True
+        self.user.save()
+        request.user = self.user
+        with self.assertRaisesMessage(PermissionDenied, ""):
+            AutocompleteJsonView.as_view(admin_site=site)(request)
+
+    def test_has_change_permission_success_with_staff_user(self):
+        factory = RequestFactory()
+        request = factory.get(self.url, {'field_identifier': 'admin_views.Answer.question', 'term': 'is'})
+        p = Permission.objects.filter(codename='change_answer')
+        self.user.user_permissions.set(p)
+        self.user.is_staff = True
+        self.user.save()
+        self.user = User.objects.get(pk=self.user.pk)
+        self.assertTrue(self.user.has_perm('admin_views.change_answer'))
+        request.user = self.user
+
+        response = AutocompleteJsonView.as_view(admin_site=site)(request)
+        self.assertEqual(response.status_code, 200)
+
+
+@override_settings(ROOT_URLCONF='admin_views.urls')
+class SeleniumTests(AdminSeleniumTestCase):
+    available_apps = ['admin_views'] + AdminSeleniumTestCase.available_apps
+
+    def setUp(self):
+        self.superuser = User.objects.create_superuser(
+            username='super', password='secret', email='super@example.com')
+        self.admin_login(username='super', password='secret', login_url=reverse('admin:index'))
+
+    def test_select(self):
+        self.selenium.get(self.live_server_url + reverse('admin:admin_views_question_add'))
+        elem = self.selenium.find_element_by_css_selector('.select2-selection')
+        elem.click()
+        results = self.selenium.find_element_by_css_selector('.select2-results')
+        self.assertTrue(results.is_displayed())
+        elem = results.find_element_by_css_selector('.select2-results__option')
+        elem.click()
+
+
+class AutocompletePaginatorTest(TestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        Question.objects.bulk_create(
+            Question(question=text_type(i))
+            for i in range(20)
+        )
+
+    def test_count(self):
+        paginator = AutocompletePaginator(Question.objects.all(), 10)
+        self.assertGreater(paginator.count, 10)
+        self.assertEqual(paginator.count, sys.maxsize)
+
+        paginator = AutocompletePaginator(Question.objects.none(), 10)
+        self.assertEqual(paginator.count, 0)
+
+    def test_validate_number(self):
+        paginator = AutocompletePaginator(Question.objects.all(), 10)
+        self.assertEqual(paginator.validate_number(1), 1)
+        self.assertEqual(paginator.validate_number(2), 2)
+        self.assertEqual(paginator.validate_number(3), 3)
+
+    def test__get_page(self):
+        paginator = AutocompletePaginator(Question.objects.all(), 10)
+        page = paginator.page(1)
+        self.assertIsInstance(page, AutocompletePage)
+
+
+class AutocompletePageTest(TestCase):
```

Comment:
Well one tests the `Paginator` and the other tests `Page`. Since these are unit tests, I feel they deserve some space.